src/rust/lib/graph-editor/src/component/node/port/output.rs

```diff
@@ -0,0 +1,300 @@
+
+            def _set_port_sizes = set_port_sizes_all.map(f!([set_port_size,data](size) {
+                let port_num = data.borrow().ports.len();
+                for index in 0..port_num {
+                    set_port_size.emit((index,*size));
+                };
+            }));
+
+            def set_port_opacity     = source::<(PortId,f32)>();
+            def set_port_opacity_all = source::<f32>();
+
+            def _set_port_opacities = set_port_opacity_all.map(f!([set_port_opacity,data](size) {
+                let port_num = data.borrow().ports.len();
+                for index in 0..port_num {
+                    set_port_opacity.emit((index,*size));
+                };
+            }));
+        }
+
+        // Init ports
+        {
+            for (index,view) in data.borrow().ports.iter().enumerate() {
+                let shape          = &view.shape;
+                let port_size    = Animation::<f32>::new(&network);
+                let port_opacity = Animation::<f32>::new(&network);
+                frp::extend! { network
+                     eval port_size.value ((size) shape.grow.set(*size));
+                     eval port_opacity.value   ((size) shape.opacity.set(*size));
+
+                    is_resize_target <- set_port_size.map(move |(id, _)| *id==index);
+                    size_change      <- set_port_size.gate(&is_resize_target);
+                    _change_size     <- size_change.map(f!(((_, size))  {
+                        port_size.set_target_value(*size)
+                    }));
+
+                    is_opacity_target <- set_port_opacity.map(move |(id, _)| *id==index);
+                    opacity_change    <- set_port_opacity.gate(&is_opacity_target);
+                    _change_size      <- opacity_change.map(f!(((_, opacity))  {
+                        port_opacity.set_target_value(*opacity)
+                    }));
+
+                    eval view.events.mouse_over ([port_size,set_port_sizes_all,port_opacity,set_port_opacity_all](_) {
+                        set_port_sizes_all.emit(BASE_SIZE);
+                        set_port_opacity_all.emit(0.5);
+                        port_size.set_target_value(HIGHLIGHT_SIZE);
+                        port_opacity.set_target_value(1.0);
+                    });
+
+                    eval view.events.mouse_out ([set_port_sizes_all,set_port_opacity_all](_) {
+                         set_port_sizes_all.emit(0.0);
+                         set_port_opacity_all.emit(0.0);
+                    });
+
+                    eval view.events.mouse_down ([frp](_) {
+                        frp.on_port_mouse_down.emit(index);
+                    });
+                }
+            }
+        }
+    }
+
+    // TODO: Implement proper sorting and remove.
+    /// Hack function used to register the elements for the sorting purposes. To be removed.
+    pub(crate) fn init_shape_order_hack(scene:&Scene) {
+        let logger = Logger::new("hack");
```

Comment:
rename it pls to `order_hack`. I knwo we have it like this in other parts, but its needs to be fixed there as well